### DISCUSSION TITLE
Please add an option to skip remove.lindep()

### DISCUSSION THREAD

**Discussion by shangzhi-hong** [amices/mice#371] [Impute, analyse and pool]
Please add option to skip the call to `remove.lindep()` in the `sampler.univ()` function.
I understand it is necessary for most models, but for methods like random-forest-based ones, it may not be necessary.
It causes errors when I do simulations, and I found it impossible to skip it.

Thanks.

**Comment by stefvanbuuren** [comment]
Did you already try something like `mice(data, maxcor = 1.0, eps = 0)`?

This will set the `maxcor` and `eps` arguments of `remove.lindep()` to its extremes, and hence all variables will pass. 

This is a dangerous option. The setting could crash `mice()` due to numerical difficulties if collinearity is not properly handled.

**Comment by gerkovink** [comment]
For many such numerical difficulties, the estimation will fall back to ridge regression for at least the default methods. If this happens, the logged events will notify the users about this expected behaviour.

**Comment by shangzhi-hong** [comment]
> Did you already try something like `mice(data, maxcor = 1.0, eps = 0)`?
> 
> This will set the `maxcor` and `eps` arguments of `remove.lindep()` to its extremes, and hence all variables will pass.
> 
> This is a dangerous option. The setting could crash `mice()` due to numerical difficulties if collinearity is not properly handled.

Thanks, I am not aware of this combination of paramters (did not see the `...` in the function call), sorry
I am aware of the consequences, but I am doing experiments with machine-learning-based methods (like CART, RF, etc.), I believe that the are robust against collinearity

**Comment by shangzhi-hong** [comment]
It seems that the `remove.lindep ` function has following lines:
```
  if (eps <= 0)
    stop("\n Argument 'eps' must be positive.")
```
that `eps = 0` would not pass this check

**Comment by shangzhi-hong** [comment]
I am wondering whether `eps = .Machine$double.xmin` or `eps = .Machine$double.eps` would do the trick

**Comment by stefvanbuuren** [comment]
`mice 3.8.4` tweaks `remove.lindep()` as follows.

```r
  # setting eps = 0 bypasses remove.lindep()
  if (eps == 0) 
    return(rep.int(TRUE, ncol(x)))
  if (eps < 0)
    stop("\n Argument 'eps' must be positive."
```

You should now be able to bypass `remove.lindep()` by setting `mice(..., eps = 0)`.

**Comment by shangzhi-hong** [comment]
Thanks, I will change my code accordingly
